parquet-hadoop/src/main/java/org/apache/parquet/crypto/FileEncryptionProperties.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+
+package org.apache.parquet.crypto;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+
+import org.apache.parquet.bytes.BytesUtils;
+import org.apache.parquet.format.EncryptionAlgorithm;
+
+public class FileEncryptionProperties {
+  
+  private final EncryptionAlgorithm algorithm;
+  private final byte[] footerKeyBytes;
+  private final byte[] footerKeyMetadata;
+  
+  private byte[] aadBytes;
+  private List<ColumnEncryptionProperties> columnList;
+  private boolean encryptTheRest;
+  //Uniform encryption means footer and all columns are encrypted, with same key
+  private boolean uniformEncryption;
+  private boolean setupProcessed;

```

Comment:
Can we somehow remove setupProcessed? I don't really like the idea that the getters modify the state of an object, and client can set some of the state via setter, but can't set the others.

How about making FileEncryptionProperties class immutable, and creating overloaded constructors for the possible construction options, or maybe even better, make all constructor private and provide a builder for this class?